### DISCUSSION TITLE
autodiscoverFilter ignored if supplied as string

### DISCUSSION THREAD

**Discussion by stevemagness** [renovatebot/renovate#18508] [Request Help]
**Potential bug** 

As part of https://github.com/renovatebot/renovate/commit/0a6d97f23a734a8bccbce78543dca2f483db0999 a change was made to make the autodiscoverFilter an array instead of a string. 

We have an automated job that runs renovatebot, always using the latest docker image. It runs every hour during the night. Last night we ran a successful job at 3:30AM using 32.241.10, but the next job at 4:30AM used 33.0.0 - it started ignoring our autodiscoveryFilter and created PRs for configuring renovateBot in a number of other projects that we did not want it to touch. We think this is due to the change made above that autodiscoverFilter is now an array. I think the logic here could be more robust; either return an error if a string is provided in the config, or convert it to an array with a single item and then everything should work as it did previously?

Note: our autodiscoveryFilter is set to a fairly simple string of the form "project/repo*"

**Comment by viceice** [comment]
you should limit the version to the major version to not silently get breaking changes.

**Comment by rarkins** [comment]
It should still support strings. How to reproduce your problem manually?

**Reply by viceice** [reply]
- https://github.com/renovatebot/renovate/issues/18529

can reproduce, string isn't migrated to array. tested from `config.js`.

**Reply by viceice** [reply]
working on a fix
